options.go

```diff
@@ -9,9 +9,18 @@ import (
 	"srcd.works/go-git.v4/plumbing/transport"
 )
 
+// SubmoduleRescursivity defines how depth will affect any submodule recursive
+// operation
+type SubmoduleRescursivity int
```

Comment:
I just change the time to uint, infinity by itself is not allowed, a number should be given always, even if is very high